### ISSUE TITLE
bios version

### ISSUE THREAD

**Issue by rudem70** [kreizlie/HP-EliteBook-840-G4-Hackintosh#7]
hi, i think that last bios for 840 g4 is P78 0135 and not BIOS Q85 01.09.01.

**Comment by galaxy-nomad** [comment]
The latest BIOS you can use with OpenCore or Clover and the EliteBook 840 G4 is P78 v1.29 Any later version will not boot. Hopefully kreizlie will chime in with an explanation why he listed the BIOS Q85 01.09.01 He may have made a mistake and was referencing another computer altogether.

**Comment by ghost** [comment]
Hi cyberfunk, 

i had the same problem, solved it by downgrading bios, was a hard way to find out and saw last chance in it, but worked very great!!! ;)

**Comment by rudem70** [comment]
> 
> 
> Hi cyberfunk,
> 
> i had the same problem, solved it by downgrading bios, was a hard way to find out and saw last chance in it, but worked very great!!! ;)

can you share bios, please. on hp support is available only last
thanks

**Comment by galaxy-nomad** [comment]
You need to downgrade one BIOS version at a time from within your BIOS settings. At least that's the way I did it. It took a while for me as mine was on the latest and I had to go back many versions. Let me know if that works for you.

**Comment by ghost** [comment]
![Screenshot 2020-10-14 at 18 59 19](https://user-images.githubusercontent.com/71799198/96021237-73075e00-0e4f-11eb-96f6-012850fba36d.png)
![Screenshot 2020-10-14 at 18 59 39](https://user-images.githubusercontent.com/71799198/96021246-7569b800-0e4f-11eb-9118-dcaa3f900302.png)
Hi rudem70, you have to click on previous versions, there you find it, for me the last working version was 1.31

**Comment by galaxy-nomad** [comment]
> ![Screenshot 2020-10-14 at 18 59 19](https://user-images.githubusercontent.com/71799198/96021237-73075e00-0e4f-11eb-96f6-012850fba36d.png)
> ![Screenshot 2020-10-14 at 18 59 39](https://user-images.githubusercontent.com/71799198/96021246-7569b800-0e4f-11eb-9118-dcaa3f900302.png)
> Hi rudem70, you have to click on previous versions, there you find it, for me the last working version was 1.31

Awesome! Yeah I didn't even think about downloading the exact version. I just hooked up an Ethernet cable and downgraded one version at a time.🤣

**Comment by ghost** [comment]
cyberfunk, try my efi ;)

**Comment by cyberfunk** [comment]
For OpenCore?

**Comment by galaxy-nomad** [comment]
> cyberfunk, try my efi ;)

Man to be honest I'm a bit hesitant to make the move to OpenCore for one reason. Using OpenCore would put me off the reservation into deep water handling things I know absolutely nothing about. RehabMan's guide/scripts for the EliteBook series basically do everything for you. OpenCore has no training wheel install scripts to my knowledge for the EliteBook 840 G4. If I used someone else's EFI I might run into issues because of differences in config.plist and other settings that I would need to study up on to know what's going on. Right now my setup is rock solid. Everything just works. Basically I'd prefer to wait for an OpenCore guide and more of a community experience before I make the switch.

**Comment by ghost** [comment]
Hi Cyberfunk you can install opencore additional next to your clover, just put it in to your efi folder next to clover and you can choose in bios by pressing f9 withouot any risk ;)

**Comment by ghost** [comment]
Cyberfunk, is your internal Wifi working?

**Comment by galaxy-nomad** [comment]
> Hi Cyberfunk you can install opencore additional next to your clover, just put it in to your efi folder next to clover and you can choose in bios by pressing f9 withouot any risk ;)

Thanks. I think eventually if I decid to upgrade to Big Sure I will do a fresh install with OpenCore as the Rehab/Clover setup comes with a lot of kexts and odds and ends installed in obscure places and I don't want any leftovers when I make the switch.

**Comment by galaxy-nomad** [comment]
> Cyberfunk, is your internal Wifi working?

Yes absolutely everything including handoff/airdrop wifi & bluetooth works with the Rehabman Clover setup on the EliteBook 840 G4. I use the Broadcom BCM94352Z Lenovo Yoga 3 Pro WiFi + BlueTooth Card. I'm running Mojave because I don't like where Apple is going with locking the file system and root of the drive in Catalina onward. Eventually I know I will have to upgrade but I'm putting it off for the time being.

**Comment by ghost** [comment]
do you have working card reader?

**Comment by galaxy-nomad** [comment]
> do you have working card reader?

I don't think so but I never use that or the pointer or fingerprint reader. I think those three things are the only things that don't work with Rehabman Guide and Clover. Are you saying you got that working in OpenCore?

**Comment by ghost** [comment]
at my github page, download the sinetek kext and put it in your clover, should work!

**Comment by galaxy-nomad** [comment]
Thanks will give it a shot! I'll have to pick up an SD card as I don't own one at the moment.

**Comment by kreizlie** [comment]
Hi guys, sorry for this late reply, I will check the BIOS version later

**Comment by ghost** [comment]
ok, i made some tests in the morning  because there is a new bios version out v.1.35 but with no success, i had to go back to 1.29/1.31
![IMG_20201025_135939](https://user-images.githubusercontent.com/71799198/97108269-aefeb680-16cc-11eb-96a0-39a8cc2203a3.jpg)

**Comment by ghost** [comment]
New BIOS causes this in DSDT RTC device:

Device (RTC) { Name (_HID, EisaId ("PNP0B00") /* AT Real-Time Clock */) // _HID: Hardware ID Name (_CRS, ResourceTemplate () // _CRS: Current Resource Settings { IO (Decode16, 0x0070, // Range Minimum 0x0070, // Range Maximum 0x01, // Alignment 0x02, // Length ) IRQNoFlags () {8} }) Method (_STA, 0, NotSerialized) // _STA: Status { If ((STAS == One)) { Return (0x0F) } Else { Return (Zero) } } }

We need to return 0x0F.
This can be fixed by ACPI rename or better way - by SSDT file. You can look at this repository.
kecinzer/hpelitebook850g5-opencore@65338fe
